### PR TITLE
Slack Notifications support

### PR DESCRIPTION
**Pull request by memory**
This is not quite working yet, but it's at the point where I need a little guidance.

### CODE REVIEW

File: src/main/java/com/palantir/stash/stashbot/servlet/RepoConfigurationServlet.java [hunk 1]
```diff
@@ -124,6 +124,27 @@ public void doGet(HttpServletRequest req, HttpServletResponse res) throws Servle
                 jenkinsServersData.add(m);
             }
 
+            // not sure if there's a better place to put this?
+            List<Map<String, String>> slackCommitInfoOptions = new ArrayList<Map<String, String>>();
+
+            HashMap<String, String> a = new HashMap<String, String>();
+            a.put("text", "nothing about commits");
+            a.put("value", "NONE");
+            a.put("selected", "true");
```
Comment (terabyte) [inline src/main/java/com/palantir/stash/stashbot/servlet/RepoConfigurationServlet.java]: You will want to figure out which is "selected" and mark that one "true", and the others false.  as-is, this doesn't persist the data and each time the page loads then is saved would reset it to the first one.

File: src/main/resources/static/repository-configuration-panel.soy [hunk 2]
```diff
@@ -264,7 +281,151 @@
               ]] /}
           {/call}
 
-          <h4>&nbsp;</h4>
+          <h4>Slack Notification Settings</h4>
```
Comment (terabyte) [inline src/main/resources/static/repository-configuration-panel.soy:284]: I have officially failed one of my new years resolutions by reading this much soy... I made it all the way to October! =)

File: src/main/java/com/palantir/stash/stashbot/config/ConfigurationPersistenceImpl.java [hunk 3]
```diff
@@ -387,7 +417,24 @@ private boolean getBoolean(HttpServletRequest req, String parameter) {
...
+                new DBParam("SLACK_CUSTOM_MESSAGE", slackSettings.getSlackCustomMessage()),
+                new DBParam("SLACK_START_NOTIFICATION", slackSettings.getSlackStartNotification()),
+                new DBParam("SLACK_NOTIFY_SUCCESS", slackSettings.getSlackNotifySuccess()),
+                new DBParam("SLACK_NOTIFY_ABORTED", slackSettings.getSlackNotifyAborted()),
+                new DBParam("SLACK_NOTIFY_NOT_BUILT", slackSettings.getSlackNotifyNotBuilt()),
+                new DBParam("SLACK_NOTIFY_UNSTABLE", slackSettings.getSlackNotifyUnstable()),
+                new DBParam("SLACK_NOTIFY_FAILURE", slackSettings.getSlackNotifyFailure()),
+                new DBParam("SLACK_NOTIFY_BACK_TO_NORMAL", slackSettings.getSlackNotifyBackToNormal()),
+                new DBParam("SLACK_NOTIFY_REPEATED_FAILURE", slackSettings.getSlackNotifyRepeatedFailure()),
+                new DBParam("SLACK_INCLUDE_TEST_SUMMARY", slackSettings.getSlackIncludeTestSummary())
```
Comment (terabyte) [inline src/main/java/com/palantir/stash/stashbot/config/ConfigurationPersistenceImpl.java:437]: It makes me sad to even utter this, but if it were me, I'd consider creating a new table and object to hold hte slack data.  This means essentially copy-pasta-ing RepositoryConfig.java and probably RepositoryConfigServlet.java and having a RepoConfig have a pointer to a SlackConfig object, etc etc.

That said, I am willing to accept this as-is, it's just a suggestion, it is a matter of taste and has its own terrible consequences.

Comment (memory) [reply src/main/java/com/palantir/stash/stashbot/config/ConfigurationPersistenceImpl.java:437]: I totally get the urge here but...oh god.  I'll check my enthusiasm level for this after I've taken a swing at the other issue but I suspect I'm gonna pass. :)

**Review by terabyte — APPROVED** [review]

**Pull request merged**